agent.go

```diff
@@ -220,6 +259,12 @@ func (s *sandbox) setContainer(id string, ctr *container) {
 
 func (s *sandbox) deleteContainer(id string) {
 	s.Lock()
+	for k := range s.storages {
+		sbs := s.storages[k]
+		sbs.Lock()
```

Comment:
This would deadlock since `DecrementRefCount` takes the lock again. And is it intentional that sbs is not removed from the map?

And what about `container.removeContainer()`?